src/analysis_and_optimization/Memory_patterns.ml

```diff
@@ -103,13 +126,16 @@ let query_stan_math_mem_pattern_support (name : string)
             |> Result.is_ok )
           namematches in
       let is_soa ((_ : UnsizedType.returntype), _, mem) =
-        mem = Mem_pattern.SoA in
+        match requested_mem with
+        | Mem_pattern.SoA -> mem = Mem_pattern.SoA || mem = OpenCL
```

Comment:
This seems like one place where the SoA and OpenCL things are considered equivalent, so I wonder if the `is` functions in `mem_pattern.ml` should do the same?

Reply:
For checking the functions `OpenCL` -> `SoA` but `SoA` != `OpenCL`